cdc/sinkv2/eventsink/txn/txn_sink.go

```diff
@@ -14,24 +14,204 @@
 package txn
 
 import (
+	"encoding/binary"
+	"hash/crc64"
+	"sync"
+
+	"github.com/pingcap/log"
 	"github.com/pingcap/tiflow/cdc/model"
 	"github.com/pingcap/tiflow/cdc/sinkv2/eventsink"
+	"github.com/pingcap/tiflow/pkg/causality"
+	"github.com/pingcap/tiflow/pkg/chann"
+	"go.uber.org/zap"
+)
+
+const (
+	defaultConflictDetectorSlots int64        = 1024 * 1024
+	crcTable                     *crc64.Table = crc64.MakeTable(crc64.ISO)
 )
 
 // Assert EventSink[E event.TableEvent] implementation
-var _ eventsink.EventSink[*model.SingleTableTxn] = (*Sink)(nil)
+var _ eventsink.EventSink[*model.SingleTableTxn] = (*sink)(nil)
+
+// sink is the sink for SingleTableTxn.
+type sink struct {
+	conflictDetector *causality.ConflictDetector[*worker, *txnEvent]
+	workers          []*worker
+}
 
-// Sink is the sink for SingleTableTxn.
-type Sink struct{}
+func newSink(backends []backend, conflictDetectorSlots int64) sink {
+	workers := make([]*worker, 0, len(backends))
+	for i, backend := range backends {
+		w := newWorker(i, backend)
+		w.wg.Add(1)
+		go w.Run()
+		workers = append(workers, w)
+	}
+	detector := causality.NewConflictDetector[*worker, *txnEvent](workers, conflictDetectorSlots)
+	return sink{detector, workers}
```

Comment:
I think naming the fields in the initialization can make the code more maintainable.